libs/design/media-gallery/src/media-gallery/media-gallery.component.ts

```diff
@@ -37,30 +39,142 @@ let uniqueGalleryId = 0;
+   * The id of the gallery.
+   */
+  @HostBinding('attr.id')
+  @Input()
+  get id() {
+    return this._id;
+  }
+  set id(val: string | undefined | null) {
+    if(!val){
+      return;
+    }
+    this._id = val;
+  };
+
   /**
    * Adds a class for styling the media gallery
+   * @docs-private
    */
   @HostBinding('class.daff-media-gallery') class = true;
 
   /**
-   * The name of the gallery
+   * The name of the gallery.
+   * @deprecated use `id` instead.
+   */
+  @Input() name: string;
+
+  /**
+   * @docs-private
+   */
+  @ContentChildren(DaffThumbnailDirective) _thumbnails: QueryList<DaffThumbnailDirective>;
+
+  /**
+   * @docs-private
    */
-  @Input() name = `${uniqueGalleryId}`;
+  @ViewChildren('thumbnailButtons', { read: ElementRef }) private _thumbnailButtons: QueryList<ElementRef<HTMLElement>>;
 
-  constructor(private registry: DaffMediaGalleryRegistry) {
+  constructor(private skeletonDirective: DaffSkeletonableDirective) {
     uniqueGalleryId++;
+    this.name = 'gallery-' + uniqueGalleryId;
+  }
+
+  get skeleton() {
+    return this.skeletonDirective.skeleton;
+  }
+
+  /**
+   * @docs-private
+   */
+  _selectedThumbnail: DaffThumbnailDirective = undefined;
+
+  /**
+   * @docs-private
+   */
+  private _selectedIndex: number = undefined;
+
+  private focusSelected() {
+    this._thumbnailButtons.get(this._selectedIndex)?.nativeElement.focus();
   }
 
-  ngOnInit() {
-    this.registry.add(this);
+  /**
+   * Select a specific entry in the media gallery by its index (starting at 0).
+   */
+  selectIndex(index: number) {
+    if(this._thumbnails.get(index)){
+      this._selectedIndex = index;
+      this._selectedThumbnail?.deselect();
+      this._thumbnails.get(index).select();
+      this._selectedThumbnail = this._thumbnails.get(index);
+    }
+  }
+
+  /**
+   * Select a specific thumbnail for this gallery.
+   */
+  selectThumbnail(thumbnail: DaffThumbnailDirective) {
+    if(!thumbnail){
+      return;
+    }
+    this._selectedThumbnail?.deselect();
+    thumbnail.select();
+    this._selectedThumbnail = thumbnail;
+    this._selectedIndex = this._thumbnails.toArray().findIndex((el) => el === thumbnail);
+  }
+
+  /**
+   * @docs-private
+   */
+  ngAfterContentInit(): void {
+    if(this._thumbnails.first) {
+      this.selectThumbnail(this._thumbnails.first);
+    }
+  }
+
+  /**
+   * Navigate to the next element in the list of thumbnails.
+   */
+  next(focus: boolean = true) {
+    this.selectIndex((this._selectedIndex + 1 + this._thumbnails.length) % this._thumbnails.length);
+    this.focusSelected();
+  }
+
+  /**
+   * Navigate to the previous element in the list of thumbnails.
+   */
+  previous(focus: boolean = true) {
+    this.selectIndex((this._selectedIndex + -1 + this._thumbnails.length) % this._thumbnails.length);
+    this.focusSelected();
   }
 
-  ngOnDestroy() {
-    this.registry.remove(this);
+  /**
+   * Select the first element
+   */
+  selectFirst(focus: boolean = true) {
```

Comment:
What's the purpose of `selectFirst` and `selectLast`? They're not used in the template, and I don't think it makes sense for users, when implementing, to dictate whether the first or last element is selected.

Reply:
This is for more complicated use cases where user's may need to programmatically select an element based upon some other kind of change in the application (page change without destroy for example).